airflow/contrib/kubernetes/worker_configuration.py

```diff
@@ -0,0 +1,371 @@
+            self.logs_volume_name: {
+                'name': self.logs_volume_name,
+                'mountPath': self.worker_airflow_logs,
+            }
+        }
+
+        if self.kube_config.dags_volume_subpath:
+            volume_mounts[self.dags_volume_name]['subPath'] = self.kube_config.dags_volume_subpath
+
+        if self.kube_config.logs_volume_subpath:
+            volume_mounts[self.logs_volume_name]['subPath'] = self.kube_config.logs_volume_subpath
+
+        if self.kube_config.dags_in_image:
+            del volumes[self.dags_volume_name]
+            del volume_mounts[self.dags_volume_name]
+
+        # Get the SSH key from secrets as a volume
+        if self.kube_config.git_ssh_key_secret_name:
+            volumes[self.git_sync_ssh_secret_volume_name] = {
+                'name': self.git_sync_ssh_secret_volume_name,
+                'secret': {
+                    'secretName': self.kube_config.git_ssh_key_secret_name,
+                    'items': [{
+                        'key': self.git_ssh_key_secret_key,
+                        'path': 'ssh',
+                        'mode': 0o440
+                    }]
+                }
+            }
+
+        if self.kube_config.git_ssh_known_hosts_configmap_name:
+            volumes[self.git_sync_ssh_known_hosts_volume_name] = {
+                'name': self.git_sync_ssh_known_hosts_volume_name,
+                'configMap': {
+                    'name': self.kube_config.git_ssh_known_hosts_configmap_name
+                },
+                'mode': 0o440
+            }
+
+        # Mount the airflow.cfg file via a configmap the user has specified
+        if self.kube_config.airflow_configmap:
+            config_volume_name = 'airflow-config'
+            config_path = '{}/airflow.cfg'.format(self.worker_airflow_home)
+            volumes[config_volume_name] = {
+                'name': config_volume_name,
+                'configMap': {
+                    'name': self.kube_config.airflow_configmap
+                }
+            }
+            volume_mounts[config_volume_name] = {
+                'name': config_volume_name,
+                'mountPath': config_path,
+                'subPath': 'airflow.cfg',
+                'readOnly': True
+            }
+
+        return volumes, volume_mounts
+
+    def generate_dag_volume_mount_path(self):
+        if self.kube_config.dags_volume_claim or self.kube_config.dags_volume_host:
+            dag_volume_mount_path = self.worker_airflow_dags
+        else:
+            dag_volume_mount_path = self.kube_config.git_dags_folder_mount_point
+
+        return dag_volume_mount_path
+
+    def make_pod(self, namespace, worker_uuid, pod_id, dag_id, task_id, execution_date,
+                 try_number, airflow_command, kube_executor_config):
+        volumes_dict, volume_mounts_dict = self._get_volumes_and_mounts()
+        worker_init_container_spec = self._get_init_containers()
+        resources = Resources(
```

Comment:
We may have added this logic or something around configuring the resources for worker task pods.